src/status_im/multiaccounts/core.cljs

```diff
@@ -10,15 +10,29 @@
             [status-im.utils.fx :as fx]
             [status-im.utils.handlers]
             [status-im.utils.gfycat.core :as gfycat]
+            [status-im.utils.identicon :as identicon]
             [status-im.utils.platform :as platform]
             [status-im.utils.utils :as utils]))
 
-(defn displayed-name [account]
-  (let [name (or (:preferred-name account) (:name account))]
+(defn displayed-name
+  "Use preferred name, name or alias in that order"
+  [{:keys [name preferred-name alias public-key]}]
+  (let [name (cond
+               (seq preferred-name) preferred-name
+               (seq name) name
+               (seq alias) alias)]
     (if (ens/is-valid-eth-name? name)
```

Comment:
so do we still calculate this for every message?

Reply:
we don't calculate it in javascript, calculation is in status-go and it's saved in the database. On new messages we always calculate it, as it's faster than pulling them from db, on loading messages it's already there, same goes from images.